benchexec/tools/nitwit.py

```diff
@@ -48,30 +47,27 @@ def version(self, executable):
     def name(self):
         return "Nitwit"
 
-    def cmdline(self, executable, options, tasks, propertyfile=None, rlimits={}):
-        if propertyfile:
-            options = options + ["-p", propertyfile]
-        return [executable] + options + tasks
+    def cmdline(self, executable, options, task, rlimits):
+        if task.property_file:
+            options = options + ["-p", task.property_file]
+        return [executable] + options + list(task.input_files_or_identifier)
 
-    def determine_result(self, returncode, returnsignal, output, isTimeout):
+    def determine_result(self, run):
         """
         See README.md at https://github.com/moves-rwth/nitwit-validator for information
         about result codes.
-        @param returncode: code returned
-        @param returnsignal: signal, which terminated validator
-        @param output: the std output
         @return: status of validator after executing a run
         """
-        if returnsignal == 0 and (returncode == 0 or returncode == 245):
+        if run.exit_code.signal is None and (run.exit_code.value == 0 or run.exit_code.value == 245):
             status = result.RESULT_FALSE_REACH
-        elif returncode is None or returncode in [-9, 9]:
+        elif run.exit_code.value is None or run.exit_code.value in [-9, 9]:
```

Comment:
Same here, plus `value` is always a positive number.
```suggestion
        elif run.exit_code.value == 9:
```

But does nitwit actually return value 9 on timeout? Does it have an internal time limit and terminates by itself is that elapses?

Reply:
I think this should have been a signal check for SIGKILL. Afaik we do not return 9 anywhere.